client/src/pages/CodeUI.js

```diff
@@ -51,15 +54,27 @@ function CodeUI(props) {
   const [question, setQuestion] = useState(" ");
   const [interview, setInterview] = useState(interviewTitle);
   const [runResult, setrunResult] = useState(null);
+  const [socket, setSocket] = useState();
+
   useEffect(() => {
     setQuestion(premadeq);
-  });
+    let socket = socketIOClient(ENDPOINT);
+    setSocket(socket);
+  }, []);
   const updateCode = (newCode) => {
     setCode(newCode);
+    socket.emit("new_code", newCode);
+    console.log("sending" + newCode);
   };
+
   const runCode = () => {
     setrunResult(eval(code));
   };
+  if (socket)
+    socket.on("update_code", (data) => {
+      console.log("updating code");
+      setCode(data);
+    });
```

Comment:
This should be in a useEffect to be honest with a dependency array for code - it's dangerous to just have this in the body of the functional component as it'll run whenever anything changes that you may not want it to run